modules/apps/site-initializer/site-initializer-raylife-ap/extra/remote-app/src/routes/policies/pages/components/Activities/TableListMobile.tsx

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayTable from '@clayui/table';
+import classnames from 'classnames';
+import React, {useEffect, useState} from 'react';
+
+import Panel from '../PolicyDetail/Panel';
+import dataActivities from './DataActivities';
+
+import './index.scss';
+
+type Props = {
+	headers: TableHeaders[];
+	rows: InfoRowContent[];
+};
+
+type TableHeaders = {
+	bold?: boolean;
+	key: string;
+	value: string;
+};
+
+const {Body, Cell, Head, Row} = ClayTable;
+
+type InfoRowContent = {[keys: string]: string};
+
+const TableListMobile: React.FC<Props> = ({headers, rows}) => {
+	const [showPanel, setShowPanel] = useState<boolean[]>([]);
+	const [selectedRow, setSelectedRow] = useState<boolean>(false);
+
+	const ContentDescription = ({
+		columnOne,
+		columnTwo,
+	}: {
+		columnOne: string;
+		columnTwo: string;
+	}) => (
+		<div className="d-flex justify-content-between ml-3 my-3">
+			<div className="d-flex font-table ml-2 mr-9">{columnOne}</div>
+
+			<div>{columnTwo}</div>
+		</div>
+	);
+
+	const isRowSelected = (index: number) => {
+		showPanel[index] === false
+			? setSelectedRow(true)
+			: setSelectedRow(false);
+	};
+
+	const displayHistoryPanel = (index: number) => {
+		const supportArray = [...showPanel];
+		isRowSelected(index);
+
+		if (selectedRow === showPanel[index].valueOf()) {
+			supportArray[index] = !supportArray[index];
+			setShowPanel(supportArray);
+		}
+		else {
+			const arrayDefault = new Array(showPanel.length).fill(false);
```

Comment:
Can this array have a more descriptive name? Is a bit confusing what it function is